src/item.h

```diff
@@ -704,8 +704,9 @@ class item : public JsonSerializer, public JsonDeserializer
         /**
          * Callback when a player starts wielding the item. The item is already in the weapon
          * slot and is called from there.
+         * @param mv number of moves *already* spent wielding the weapon
          */
-        void on_wield( player &p );
+        void on_wield( player &p, int mv = 0 );
```

Comment:
Right, `wield_contents` has its own additional move cost. Maybe it should just call `wield` and let it do the work? This would also get rid of the duplicated code for dropping the current weapon vs. putting it into the inventory.

The check for whether it's a worn item, in worn container or from main inventory gets a bit complicated, through. Tough questions. That's why I'm not sure what's better. The current solution is probably good enough (given the complicate situation). And it's documented, so there shouldn't be a problem.